src/Entity/SectionAssociation.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Drupal\workbench_access\Entity;
+
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+
+/**
+ * Defines the workbench_access SectionAssociation class.
+ *
+ * @ContentEntityType(
+ *   id = "section_association",
+ *   label = @Translation("Section association"),
+ *   bundle_label = @Translation("Section association"),
+ *   handlers = {
+ *     "access" = "Drupal\workbench_access\SectionAssociationAccessControlHandler",
+ *     "storage" = "Drupal\workbench_access\SectionAssociationStorage",
+ *     "storage_schema" = "Drupal\workbench_access\SectionAssociationStorageSchema",
+ *     "views_data" = "\Drupal\views\EntityViewsData",
+ *     "form" = {
+ *       "default" = "Drupal\workbench_access\SectionAssociationForm"
+ *     }
+ *   },
+ *   admin_permission = "assign workbench access",
+ *   base_table = "section_association",
+ *   data_table = "section_association_data",
+ *   translatable = FALSE,
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "uuid" = "uuid",
+ *   }
+ * )
+ */
+class SectionAssociation extends ContentEntityBase implements SectionAssociationInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
+
+    // Standard field, used as unique if primary index.
+    $fields['id'] = BaseFieldDefinition::create('integer')
+      ->setLabel(t('ID'))
+      ->setDescription(t('The ID of the Section Association entity.'))
+      ->setReadOnly(TRUE);
+
+    // Standard field, unique outside of the scope of the current project.
+    $fields['uuid'] = BaseFieldDefinition::create('uuid')
+      ->setLabel(t('UUID'))
+      ->setDescription(t('The UUID of the Section Association entity.'))
+      ->setReadOnly(TRUE);
+
+    // Assigned users.
+    $fields['user_id'] = BaseFieldDefinition::create('entity_reference')
```

Comment:
This should be called users because it is not an ID anymore. It is an entity. Also `->setCardinality` unlimited is still missing.